src/extension/background.js

```diff
@@ -59,6 +59,11 @@ chrome.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
   if (tab && tab.url && typeof internalActions[actionFromTab] === 'function') {
     resp = await internalActions[actionFromTab](message, sender);
   }
+
+  if (actionFromTab === 'updateUI') {
+    checkTab(tab.id);
```

Comment:
Not sure about this, seem to work tho and doesn't effect the current state of page DOM by calling `checkTab` again.

Reply:
Why do you need to update the UI when the display state changes?

Reply:
To update the extension icon in the browser